Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -54,9 +55,14 @@ namespace ROS2
         return "world";
     }
 
+    bool ROS2FrameComponent::IsTopLevel() const
+    {
+        return GetGlobalFrameName() == GetParentFrameID();
+    }
+
     bool ROS2FrameComponent::IsDynamic() const
     {   // TODO - determine by joint type
-        return GetGlobalFrameName() == GetParentFrameID();
+        return IsTopLevel();
```

Comment:
Not a fan of this assumption. Stationary robots also exist (i.e. the root of a stationary manipulator - base-link - in a warehouse map).

Reply:
Sure, but this will be determined based on joint type as in TODO - and also perhaps on another type of description. Not for now, for later